apps/astarte_appengine_api/lib/astarte_appengine_api/device/queries.ex

```diff
@@ -683,41 +696,28 @@ defmodule Astarte.AppEngine.API.Device.Queries do
     with {:ok, result} <- Repo.fetch(query, device_id, opts),
          {:ok, alias_value} <- get_value(result, alias_tag, :alias_tag_not_found),
          :ok <- check_alias_ownership(keyspace, device_id, alias_tag, alias_value) do
-      delete_alias_from_device_statement = """
-      DELETE aliases[:alias_tag]
-      FROM devices
-      WHERE device_id = :device_id
-      """
-
-      delete_alias_from_device_query =
-        DatabaseQuery.new()
-        |> DatabaseQuery.statement(delete_alias_from_device_statement)
-        |> DatabaseQuery.put(:alias_tag, alias_tag)
-        |> DatabaseQuery.put(:device_id, device_id)
-        |> DatabaseQuery.consistency(:each_quorum)
-        |> DatabaseQuery.convert()
-
-      delete_alias_from_names_statement = """
-      DELETE FROM names
-      WHERE object_name = :alias AND object_type = 1
-      """
-
-      delete_alias_from_names_query =
-        DatabaseQuery.new()
-        |> DatabaseQuery.statement(delete_alias_from_names_statement)
-        |> DatabaseQuery.put(:alias, alias_value)
-        |> DatabaseQuery.put(:device_id, device_id)
-        |> DatabaseQuery.consistency(:each_quorum)
-        |> DatabaseQuery.convert()
-
-      delete_batch =
-        CQEx.cql_query_batch(
-          consistency: :each_quorum,
-          mode: :logged,
-          queries: [delete_alias_from_device_query, delete_alias_from_names_query]
+      map_new_alias = MapSet.new([alias_tag])
+
+      query_delete_alias =
+        from ApiDevice,
+          prefix: ^keyspace,
+          where: [device_id: ^device_id],
+          update: [set: [aliases: fragment("aliases - ?", ^map_new_alias)]]
+
+      sql_query_delete_alias = Repo.to_sql(:update_all, query_delete_alias)
+
+      query_delete_in_name =
+        from(d in Name,
+          prefix: ^keyspace,
+          where: [object_name: ^alias_value, object_type: 1]
         )
 
-      with {:ok, _result} <- DatabaseQuery.call(client, delete_batch) do
+      sql_query_delete_in_name = Repo.to_sql(:delete_all, query_delete_in_name)
+
+      update_and_delete_batch =
+        %Exandra.Batch{queries: [sql_query_delete_alias, sql_query_delete_in_name]}
+
+      with :ok <- Exandra.execute_batch(Repo, update_and_delete_batch, consistency: :each_quorum) do
         :ok
       else
         %{acc: _, msg: error_message} ->
```

Comment:
```suggestion
      Exandra.execute_batch(Repo, update_and_delete_batch, consistency: :each_quorum)
```

apart from the fact that execute_batch returns [:ok or {:error, reason}](https://hexdocs.pm/exandra/0.10.2/Exandra.html#execute_batch/3) and not a map, the with can be removed as the returned value is the same